mlrun/api/utils/builder.py

```diff
@@ -41,9 +45,41 @@ def make_dockerfile(
     extra: str = "",
     user_unix_id: int = None,
     enriched_group_id: int = None,
+    builder_env: typing.List[client.V1EnvVar] = None,
+    project_secrets: dict = None,
+    extra_args: str = "",
 ):
```

Comment:
add function docstring with purpose for each argument

in general, i dont see a reason why kaniko pod arguments needs to be ENV x=y